src/utils.js

```diff
@@ -873,43 +873,34 @@ export function toIdSchema(
 
 export function toPathSchema(schema, name = "", definitions, formData = {}) {
   const pathSchema = {
-    $name: name,
+    $name: name.replace(/\.$/, ""),
   };
   if ("$ref" in schema || "dependencies" in schema) {
     const _schema = retrieveSchema(schema, definitions, formData);
     return toPathSchema(_schema, name, definitions, formData);
   }
-  if ("items" in schema) {
-    const retVal = {};
-    if (Array.isArray(formData) && formData.length > 0) {
-      formData.forEach((element, index) => {
-        retVal[`${index}`] = toPathSchema(
-          schema.items,
-          `${name}.${index}`,
-          definitions,
-          element
-        );
-      });
+  if (
+    schema.hasOwnProperty("items") &&
+    Array.isArray(formData) &&
+    formData.length > 0
+  ) {
+    formData.forEach((element, i) => {
+      pathSchema[i] = toPathSchema(
+        schema.items,
+        `${name}${i}.`,
+        definitions,
+        element
+      );
+    });
+  } else if (schema.hasOwnProperty("properties")) {
+    for (const property in schema.properties || {}) {
```

Comment:
Can we just make this:

```suggestion
    for (const property in schema.properties) {
```

Reply:
I think that should be fine. I'll check it out.